voltdbclient/client.go

```diff
@@ -0,0 +1,240 @@
+		return nil, fmt.Errorf("Can not call procedure on closed Client.")
+	}
+	handle := atomic.AddInt64(&client.clientHandle, 1)
+	cb := client.netListener.registerCallback(handle)
+	if err := client.writeProcedureCall(client.writer, procedure, handle, params); err != nil {
+		client.netListener.removeCallback(handle)
+		return nil, err
+	}
+	return <-cb.Channel, nil
+}
+
+// CallAsync asynchronously invokes the procedure 'procedure' with parameter values 'params'.
+// A pointer to the Response from the server will be put on the returned channel.
+func (client *Client) CallAsync(procedure string, params ...interface{}) (*Callback, error) {
+	if client.writer == nil {
+		return nil, fmt.Errorf("Can not call procedure on closed Client.")
+	}
+	handle := atomic.AddInt64(&client.clientHandle, 1)
+	cb := client.netListener.registerCallback(handle)
+	if err := client.writeProcedureCall(client.writer, procedure, handle, params); err != nil {
+		client.netListener.removeCallback(handle)
+		return nil, err
+	}
+	return cb, nil
+}
+
+func (client *Client) CreateConnection(hostAndPort string) error {
+	raddr, err := net.ResolveTCPAddr("tcp", hostAndPort)
+	if err != nil {
+		return fmt.Errorf("Error resolving %v.", hostAndPort)
+	}
+	var tcpConn *net.TCPConn
+	if tcpConn, err = net.DialTCP("tcp", nil, raddr); err != nil {
+		return err
+	}
+	client.reader = tcpConn
+	client.writer = tcpConn
+	login, err := serializeLoginMessage(client.config.username, client.config.password)
+	if err != nil {
+		return err
+	}
+	client.writeLoginMessage(&login)
+	if client.connData, err = client.readLoginResponse(); err != nil {
+		return err
+	}
+	client.clientHandle = 0
+	client.netListener = NewListener(client.reader)
+	client.netListener.start()
+	return nil
+}
+
+// Close a client if open. A Client, once closed, has no further use.
+// To open a new client, use NewClient.
+func (client *Client) Close() error {
+	var err error = nil
+	if client.reader != nil {
+		tcpConn := client.reader.(*net.TCPConn)
+		tcpConn.Close()
+	}
+	client.reader = nil
+	client.writer = nil
+	client.connData = nil
+	return err
+}
+
+// GoString provides a default printable format for Client.
+func (client *Client) GoString() string {
+	if client.connData != nil {
+		return fmt.Sprintf("hostId:%v, connId:%v, leaderAddr:%v buildString:%v",
+			client.connData.hostId, client.connData.connId,
+			client.connData.leaderAddr, client.connData.buildString)
+	}
+	return "uninitialized"
+}
+
+// MultiplexCallbacks 'fans in' callbacks - listens for the given set of callbacks on one channel
+func (client *Client) MultiplexCallbacks(callbacks []*Callback) <-chan *Response {
+	c := make(chan *Response)
+	for _, callback := range callbacks {
+		ch := callback.Channel
+		go func() {
+			c <- <-ch
+		}()
+	}
+	return c
+}
+
+// Ping the database for liveness.
+func (client *Client) PingConnection() bool {
+	if client.writer == nil {
+		return false
+	}
+	rsp, err := client.Call("@Ping")
```

Comment:
This is done under the hood in async manner in java how can this be made hidden for users?